pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -32,6 +32,30 @@ type ControlPlaneClientImpl struct {
 	// hostedControlPlaneNamespaceName is the namespace name on the management cluster
 	// on which the control-plane objects reside
 	hostedControlPlaneNamespaceName string
+
```

Comment:
this is well intentioned, but two mode of operations create room for bugs. We need to evaluate the opportunity cost here

Reply:
after some (admittedly not too much) more reflection, I think this is well intentioned but too risky. Test code is meant to exercise the production code as much as possible and this is a nice idea but make us diverge too much. I'm already a bit concerned about the previous comment (which gets `Status` from configmap and transparently merges into the main object), this one feels far too above the threshold.

I think we should accept the performance cost for the time being.

Reply:
wait, I misunderstood the code. This really ties to the previous commit and basically enables the API backward  compatibility to restore the status in the same perfprof object. Let me rethink.